src/sentry/integrations/issues.py

```diff
@@ -194,6 +195,38 @@ def get_issue_display_name(self, external_issue):
         """
         return ''
 
+    def get_repository_choices(self, group, **kwargs):
+        """
+        Returns the default repository and a set/subset of repositories of asscoaited with the installation
+        """
+        try:
+            repos = self.get_repositories()
+        except ApiError:
+            repo_choices = [(' ', ' ')]
+        else:
+            repo_choices = [(repo['identifier'], repo['name']) for repo in repos]
+
+        params = kwargs.get('params', {})
+        defaults = self.get_project_defaults(group.project_id)
+        default_repo = params.get('repo', defaults.get('repo') or repo_choices[0][0])
+
+        # If a repo has been selected outside of the default list of
+        # repos, stick it onto the front of the list so that it can be
+        # selected.
+        try:
+            next(True for r in repo_choices if r[0] == default_repo)
+        except StopIteration:
+            repo_choices.insert(0, self.create_default_repo_choice(default_repo))
+
+        return default_repo, repo_choices
+
+    def create_default_repo_choice(self, default_repo):
```

Comment:
Are there other implementations that can benefit from these new abstractions?

Reply:
@markstory Yes. So any integration with repos. I'll have to do this with vsts, bitbucket, and gitlab. I thought this would be a better way instead of more/less copy paste

Reply:
Sounds good to me.